src/commands/project/retrieve/start.ts

```diff
@@ -163,31 +164,40 @@ export default class RetrieveMetadata extends SfCommand<RetrieveResultJson> {
     const format: Format = flags['target-metadata-dir'] ? 'metadata' : 'source';
     const zipFileName = flags['zip-file-name'] ?? DEFAULT_ZIP_FILE_NAME;
 
-    // eslint-disable-next-line @typescript-eslint/require-await
-    Lifecycle.getInstance().on('apiVersionRetrieve', async (apiData: RetrieveVersionData) => {
-      this.log(
-        messages.getMessage('apiVersionMsgDetailed', [
-          'Retrieving',
-          apiData.manifestVersion,
-          flags['target-org'].getUsername(),
-          apiData.apiVersion,
-        ])
-      );
-    });
-
     this.spinner.start(messages.getMessage('spinner.start'));
 
     const { componentSetFromNonDeletes, fileResponsesFromDelete = [] } = await buildRetrieveAndDeleteTargets(
       flags,
       format
     );
+    if (flags.manifest || flags.metadata) {
+      const access = new RegistryAccess();
+      if (wantsToRetrieveCustomFields(componentSetFromNonDeletes, access)) {
+        this.warn(messages.getMessage('wantsToRetrieveCustomFields'));
+        componentSetFromNonDeletes.add({
+          fullName: ComponentSet.WILDCARD,
+          type: access.getTypeByName('CustomObject'),
+        });
+      }
+    }
     const retrieveOpts = await buildRetrieveOptions(flags, format, zipFileName, resolvedTargetDir);
 
     this.spinner.status = messages.getMessage('spinner.sending');
 
     this.retrieveResult = new RetrieveResult({} as MetadataApiRetrieveStatus, componentSetFromNonDeletes);
 
     if (componentSetFromNonDeletes.size !== 0 || retrieveOpts.packageOptions?.length) {
+      // eslint-disable-next-line @typescript-eslint/require-await
```

Comment:
you *could* 

`        return Promise.resolve();`
at the bottom of the function instead of doing the eslint comment.  🤷🏻